### DISCUSSION TITLE
Forwarding the audio from the browser and playing it in the scrcpy

### DISCUSSION THREAD

**Discussion by AmalVR** [yume-chan/ya-webadb#664] [Q&A]
I am working on modifying the Tango project to capture media from the browser, stream it to the scrcpy-server, and inject the audio using Android's AudioTrack. I’m referring to the sample implementation provided by @yume-chan 
[https://github.com/Genymobile/scrcpy/issues/3880#issuecomment-1595722119](scrcpy#3880).

The audio format received from browsers varies; for example, Chrome provides audio in the audio/webm; codecs=opus format, while the Android AudioTrack expects PCM. Could someone suggest a library or sample that can extract Opus audio from the WebM container and convert it to PCM for use with Android's AudioTrack? Is it possible to use MediaCodec in Android to decode and feed the audio to AudioTrack?

**Comment by yume-chan** [comment]
I'm not an Android developer, but since Scrcpy uses MediaCodec API to encode PCM audio into opus, I think it's also possible to do the opposite.

**Reply by AmalVR** [reply]
You are right, we were able to achieve that.
Thanks!
